exporter/sumologicexporter/carbon_formatter.go

```diff
@@ -0,0 +1,108 @@
+// In addition, metric name and unit are also included
+// In case `metric` or `unit` attributes has been set too, they are prefixed with underscore `_`
+func carbon2TagString(record metricPair) string {
+	length := record.attributes.Len()
+
+	if _, ok := record.attributes.Get("metric"); ok {
+		length++
+	}
+
+	if _, ok := record.attributes.Get("unit"); ok && len(record.metric.Unit()) > 0 {
+		length++
+	}
+
+	returnValue := make([]string, 0, length)
+	record.attributes.ForEach(func(k string, v pdata.AttributeValue) {
+		if k == "name" || k == "unit" {
+			k = fmt.Sprintf("_%s", k)
+		}
+		returnValue = append(returnValue, fmt.Sprintf("%s=%s", k, tracetranslator.AttributeValueToString(v, false)))
+	})
+
+	returnValue = append(returnValue, fmt.Sprintf("metric=%s", record.metric.Name()))
+
+	if len(record.metric.Unit()) > 0 {
+		returnValue = append(returnValue, fmt.Sprintf("unit=%s", record.metric.Unit()))
+	}
+
+	return strings.Join(returnValue, " ")
+}
+
+// carbon2IntRecord converts IntDataPoint to carbon2 metric string
+// with additional information from metricPair
+func carbon2IntRecord(record metricPair, dataPoint pdata.IntDataPoint) string {
+	return fmt.Sprintf("%s  %d %d",
+		carbon2TagString(record),
+		dataPoint.Value(),
+		dataPoint.Timestamp()/1e9,
+	)
+}
+
+// carbon2DoubleRecord converts DoubleDataPoint to carbon2 metric string
+// with additional information from metricPair
+func carbon2DoubleRecord(record metricPair, dataPoint pdata.DoubleDataPoint) string {
+	return fmt.Sprintf("%s  %g %d",
+		carbon2TagString(record),
+		dataPoint.Value(),
+		dataPoint.Timestamp()/1e9,
+	)
+}
+
+// carbon2metric2String converts metric to Carbon2 formatted string
+func carbon2Metric2String(record metricPair) string {
+	var nextLines []string
+
+	switch record.metric.DataType() {
+	case pdata.MetricDataTypeIntGauge:
+		for i := 0; i < record.metric.IntGauge().DataPoints().Len(); i++ {
+			dataPoint := record.metric.IntGauge().DataPoints().At(i)
+			nextLines = append(nextLines, carbon2IntRecord(record, dataPoint))
+		}
+	case pdata.MetricDataTypeIntSum:
+		for i := 0; i < record.metric.IntSum().DataPoints().Len(); i++ {
+			dataPoint := record.metric.IntSum().DataPoints().At(i)
+			nextLines = append(nextLines, carbon2IntRecord(record, dataPoint))
+		}
+	case pdata.MetricDataTypeDoubleGauge:
+		for i := 0; i < record.metric.DoubleGauge().DataPoints().Len(); i++ {
+			dataPoint := record.metric.DoubleGauge().DataPoints().At(i)
+			nextLines = append(nextLines, carbon2DoubleRecord(record, dataPoint))
+		}
+	case pdata.MetricDataTypeDoubleSum:
+		for i := 0; i < record.metric.DoubleSum().DataPoints().Len(); i++ {
+			dataPoint := record.metric.DoubleSum().DataPoints().At(i)
+			nextLines = append(nextLines, carbon2DoubleRecord(record, dataPoint))
+		}
+	// Skip complex metrics
+	case pdata.MetricDataTypeDoubleHistogram:
+	case pdata.MetricDataTypeIntHistogram:
+	case pdata.MetricDataTypeDoubleSummary:
+	}
+
+	return strings.Join(nextLines, "\n")
```

Comment:
Take it or leave it suggestiong. This way if there are more data points we save some allocations.

```suggestion
	var nextLines []string

	switch record.metric.DataType() {
	case pdata.MetricDataTypeIntGauge:
		dps := record.metric.IntGauge().DataPoints()
		nextLines = make([]string, 0, dps.Len())
		for i := 0; i < dps.Len(); i++ {
			nextLines = append(nextLines, carbon2IntRecord(record, dps.At(i)))
		}
	case pdata.MetricDataTypeIntSum:
		dps := record.metric.IntSum().DataPoints()
		nextLines = make([]string, 0, dps.Len())
		for i := 0; i < dps.Len(); i++ {
			nextLines = append(nextLines, carbon2IntRecord(record, dps.At(i)))
		}
	case pdata.MetricDataTypeDoubleGauge:
		dps := record.metric.DoubleGauge().DataPoints()
		nextLines = make([]string, 0, dps.Len())
		for i := 0; i < dps.Len(); i++ {
			nextLines = append(nextLines, carbon2DoubleRecord(record, dps.At(i)))
		}
	case pdata.MetricDataTypeDoubleSum:
		dps := record.metric.DoubleSum().DataPoints()
		nextLines = make([]string, 0, dps.Len())
		for i := 0; i < dps.Len(); i++ {
			nextLines = append(nextLines, carbon2DoubleRecord(record, dps.At(i)))
		}
	// Skip complex metrics
	case pdata.MetricDataTypeDoubleHistogram:
	case pdata.MetricDataTypeIntHistogram:
	case pdata.MetricDataTypeDoubleSummary:
	}

	return strings.Join(nextLines, "\n")
```